src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -651,10 +651,10 @@ private DependencyGraphSpec GetDependencyGraphSpec(string entryProjectPath, IDic
                 // Load the projects via MSBuild and create an array of them since Parallel.ForEach is optimized for arrays
                 var projects = LoadProjects(entryProjects)?.ToArray();
 
-                // If no projects were loaded, return null indicating that the projects could not be loaded.
+                // If no projects were loaded, return an empty DependencyGraphSpec
                 if (projects == null || projects.Length == 0)
                 {
-                    return null;
+                    return new DependencyGraphSpec();
```

Comment:
Why do we need tihs change?

Reply:
This method returns `null` if there's an exception after it logs an error but in the case where you have a solution file with no projects to restore, a message is logged and this returns `null` but that ends up making the EXE return an exit code of `1` and you get an error message that static graph-based restore returned `1` but didn't log an error.  I checked regular restore and it just reports success.  So returning an empty `DependencyGraphSpec` makes more sense because now the caller can tell if it wasn't possible to get one (because of an error) or if there are just no projects to restore.  Now the task logs no error but reports success and there's a message indicating why:

Before:
```
Restore:
  Determining projects to restore...
  The solution contains '1' project(s) 'a' that are not known to MSBuild. Ensure that all projects are known to be MSBuild before running restore on the solution.
  Evaluated 0 project(s) in 326ms (0 builds, 0 failures).
D:\nuget3\artifacts\NuGet.Build.Tasks.Console\bin\Debug\net472\NuGet.RestoreEx.targets(19,5): error : Static graph-based restore failed with exit code '1' but did not log an error. Please file an issue at https://github
.com/NuGet/Home. [D:\nuget3\.test\work\e3ff513d\04b28e2f\solution\solution.sln]

Build FAILED.

"D:\nuget3\.test\work\e3ff513d\04b28e2f\solution\solution.sln" (Restore target) (1) ->
(Restore target) ->
  D:\nuget3\artifacts\NuGet.Build.Tasks.Console\bin\Debug\net472\NuGet.RestoreEx.targets(19,5): error : Static graph-based restore failed with exit code '1' but did not log an error. Please file an issue at https://gith
ub.com/NuGet/Home. [D:\nuget3\.test\work\e3ff513d\04b28e2f\solution\solution.sln]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:58.55
```

After:
```
Restore:
  Determining projects to restore...
  The solution contains '1' project(s) 'a' that are not known to MSBuild. Ensure that all projects are known to be MSBuild before running restore on the solution.
  Evaluated 0 project(s) in 196ms (0 builds, 0 failures).
  The solution did not have any projects to restore, ensure that all projects are known to be MSBuild and that the projects exist.

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.66
```

The new behavior matches regular restore now!